[docs] Explain how to disable Base Select's portal (#37684)
mui/material-ui: docs/data/base/components/select/select.md

<Select slotProps={{ listbox: { className: 'my-listbox' } }} />
```

### Portals

> By default, the Select's popup is rendered in a portal and appended to the bottom of the DOM.

Reviewer: ```suggestion
By default, the Select's popup is rendered in a [portal](https://mui.com/base-ui/react-portal/components-api/) and appended to the bottom of the DOM.
```

i think it might be good to add link to portal here, it will be helpful for people who are starting with react as they might not know what portal means.

> If you wish to render the popup where the component is defined, override the `disablePortal` prop of the underlying Popper:

Reviewer 2: Copyediting nitpicks :)

```suggestion
To instead render the popup where the component is defined, override the `disablePortal` prop of the underlying Popper, as shown below:
```
